cmd/thanos/receive.go

```diff
@@ -291,6 +291,22 @@ func runReceive(
 		)
 	}
 
+	level.Debug(logger).Log("msg", "setting up periodic tenant pruning")
+	{
+		ctx, cancel := context.WithCancel(context.Background())
+		g.Add(func() error {
+			return runutil.Repeat(2*time.Hour, ctx.Done(), func() error {
```

Comment:
Is there a reason for this interval choice? I'm wondering if it would be beneficial to document the choice.

Reply:
It's really arbitrary and more of an implementation detail. I though it shouldn't be too often to avoid lock contention, but also more frequent than a typical retention period. Two hours seems to be the solution for everything related to Prometheus :)

Reply:
You're right, I was thinking what would be benefits of having longer period or having this configurable, but there are none that I can think of. I guess the magical two hour period is a reasonable choice :slightly_smiling_face:  Thanks!